segment.go

```diff
@@ -168,3 +168,14 @@ type DocVisitState interface {
 type StatsReporter interface {
 	ReportBytesWritten(bytesWritten uint64)
 }
+
+type FieldStatsReporter interface {
+	UpdateFieldStats(FieldStats)
+}
+
+type FieldStats interface {
+	AddStat(statName, fieldName string, value uint64)
```

Comment:
Drop the word `Stat` for all these member methods, the interface already has that word.

Also let's rename ..
* AddStat -> Store(..)
* AggregateStats -> Aggregate(..)
* GetStatsMap -> Fetch()